Software/Human-Zombie-Device/Human-Zombie-Device.ino

```diff
@@ -29,52 +26,54 @@ void setup() {
   }
   Serial.println("LoRa Initializing OK!");
 
+
   // Get the saved state from EEPROM and output it
   deviceState = init_and_read_device_state();
+  numberOfInfectionTicks = 0; // set the infection counter to zero. 
   write_oled_and_serial_line( 2, DeviceStateToString(deviceState) );
+
+
 }
 
 void loop() {
 
   if (is_message_ztag( get_new_lora_packet() ) ) {
 
-    Serial.println("ZTAg Packet received, doing switch now");
     DeviceState receivedState = StringToDeviceState(receivedPacket.message);
-    // Main behaviour loop
+    // Main behaviour loop, only entered if we find a ZTag message
     if ( DeviceState::human == deviceState ){
-      Serial.println("Currently Human, investigating new packet");
+
       if (DeviceState::infected == receivedState ) {
-        deviceState = DeviceState::infected;
-        Serial.println("INFECTED!!! Going to infected state");
-        write_oled_and_serial_line( 2, DeviceStateToString(deviceState) );
+        is_infected_yet(receivedPacket.rssi);
       }
-      Serial.println("Done in Human state checking");
+
+      //TODO: Decision time, currently we don't reset the health bar if you go in a medical
+      // zone whilst a human. Maybe we should. If we should then set to medicalzone here.
```

Comment:
Maybe a "Slow Recovery back to Health" instead of an "Instant reset"

Reply:
maybe!